test/e2e/tests/send-hex-address.spec.js

```diff
@@ -123,43 +123,20 @@ describe('Send ERC20 to a 40 character hexadecimal address', function () {
     await withFixtures(
       {
         dapp: true,
-        fixtures: new FixtureBuilder()
-          .withPermissionControllerConnectedToTestDapp()
-          .build(),
+        fixtures: new FixtureBuilder().withTokensControllerERC20().build(),
         ganacheOptions,
         smartContract,
         title: this.test.title,
         failOnConsoleError: false,
       },
-      async ({ driver, contractRegistry }) => {
-        const contractAddress = await contractRegistry.getContractAddress(
-          smartContract,
-        );
+      async ({ driver }) => {
         await driver.navigate();
         await driver.fill('#password', 'correct horse battery staple');
         await driver.press('#password', driver.Key.ENTER);
-
-        await driver.openNewPage(
-          `http://127.0.0.1:8080/?contract=${contractAddress}`,
-        );
-        let windowHandles = await driver.getAllWindowHandles();
-        const extension = windowHandles[0];
-
-        // Using the line below to make wait time deterministic and avoid using a delay
-        // See more here https://github.com/MetaMask/metamask-extension/pull/15604/files#r949300551
-        await driver.findClickableElement('#deployButton');
-
-        // Add token
-        await driver.clickElement('#watchAsset');
-        await driver.waitUntilXWindowHandles(3);
-        windowHandles = await driver.getAllWindowHandles();
-        await driver.switchToWindowWithTitle(
-          'MetaMask Notification',
-          windowHandles,
-        );
-        await driver.clickElement({ text: 'Add token', tag: 'button' });
-        await driver.waitUntilXWindowHandles(2);
-        await driver.switchToWindow(extension);
+        await driver.waitForSelector({
+          css: '[data-testid="eth-overview__primary-currency"]',
+          text: '24.9977 ETH',
```

Comment:
If we pass the ganache provider to the test suite, we could avoid hardcoding the balance and directly grab it from ganache. This way, won't have to update the balance anymore in the future, in case some more tx's / deployments happen before the test.

It will look something like this. In `helpers.js` file, we could pass the provider to the testSuite:

```
L59     const provider = ganacheServer.getProvider();

....
L 144     await testSuite({
                  driver: driverProxy ?? driver,
                  mockServer,
                  contractRegistry,
                  provider,
                });
```

And then on the testcase we can use the provider function for getting the balance:
```
L197      async ({ driver, provider }) => {
                 await driver.navigate();
                await driver.fill('#password', 'correct horse battery staple');
                await driver.press('#password', driver.Key.ENTER);

                const accounts = await provider.request({ method: "eth_accounts", params: [] });
                const balanceHex = await provider.request({ method: "eth_getBalance", params: [accounts[0], "latest"] });
                const balanceInt = (parseInt(balanceHex, 16)) / 10 ** 18;

                await driver.waitForSelector({
                  css: '[data-testid="eth-overview__primary-currency"]',
                  text: `${balanceInt.toFixed(4)} ETH`,
                });
```

Reply:
Good suggestion. 
We can explore this in a separate PR since we are also waiting for a balance to update in multiple other tests.

Reply:
sounds a good idea! Thank you!!